src/GraphQL/Introspection/ISchemaComparer.cs

```diff
@@ -0,0 +1,134 @@
+        /// If this returns <see langword="null"/> then the original field ordering is preserved.
+        /// </summary>
+        /// <param name="parent"> Parent type to which the fields belong. </param>
+        IComparer<IFieldType> FieldComparer(IGraphType parent);
+
+        /// <summary>
+        /// Returns a comparer for field arguments.
+        /// If this returns <see langword="null"/> then the original argument ordering is preserved.
+        /// </summary>
+        /// <param name="field"> The field to which the arguments belong. </param>
+        IComparer<QueryArgument> ArgumentComparer(IFieldType field);
+
+        /// <summary>
+        /// Returns a comparer for enum values.
+        /// If this returns <see langword="null"/> then the original enum value ordering is preserved.
+        /// </summary>
+        /// <param name="parent"> The enumeration to which the enum values belong. </param>
+        IComparer<EnumValueDefinition> EnumValueComparer(EnumerationGraphType parent);
+
+        /// <summary>
+        /// Returns a comparer for GraphQL directives.
+        /// If this returns <see langword="null"/> then the original directive ordering is preserved.
+        /// </summary>
+        IComparer<DirectiveGraphType> DirectiveComparer { get; }
+    }
+
+    /// <summary>
+    /// Default schema comparer. By default all elements are returned as is, no sorting is applied.
+    /// </summary>
+    public class DefaultSchemaComparer : ISchemaComparer
+    {
+        /// <inheritdoc/>
+        public virtual IComparer<IGraphType> TypeComparer => null;
+
+        /// <inheritdoc/>
+        public virtual IComparer<DirectiveGraphType> DirectiveComparer => null;
+
+        /// <inheritdoc/>
+        public virtual IComparer<QueryArgument> ArgumentComparer(IFieldType field) => null;
+
+        /// <inheritdoc/>
+        public virtual IComparer<EnumValueDefinition> EnumValueComparer(EnumerationGraphType parent) => null;
+
+        /// <inheritdoc/>
+        public virtual IComparer<IFieldType> FieldComparer(IGraphType parent) => null;
+    }
+
+    /// <summary>
+    /// Default schema comparer for GraphQL.NET v3.x.x.
+    /// By default only fields are ordered by their names within enclosing type.
+    /// </summary>
+    public sealed class LegacyV3SchemaComparer : DefaultSchemaComparer
+    {
+        private static readonly FieldByNameComparer _instance = new FieldByNameComparer();
+
+        private sealed class FieldByNameComparer : IComparer<IFieldType>
+        {
+            public int Compare(IFieldType x, IFieldType y) => x.Name.CompareTo(y.Name);
+        }
+
+        /// <inheritdoc/>
+        public override IComparer<IFieldType> FieldComparer(IGraphType parent) => _instance;
+    }
+
+    /// <summary>
+    /// All elements are sorted in alphabetical order of their names.
+    /// </summary>
+    public class AlphabeticallySchemaComparer : ISchemaComparer
```

Comment:
I'm not sure about naming - _Alphabetically_.

Reply:
I agree.  Don't have any better suggestions.  Perhaps `AlphabeticalSchemaComparer` ?

Reply:
Done